forc-plugins/forc-doc/src/lib.rs

```diff
@@ -108,7 +107,8 @@ pub fn compile_html(
         tests_enabled,
         &engines,
         None,
-        experimental,
+        &[],
```

Comment:
Just a thought, something we didn't discuss. We will not have documentation for experimental features available. Not an issue IMO.

Reply:
Not sure I understood.

Reply:
Hmm, after giving it a second thought I think I was too optimistic. Essentially, `forc doc` needs to be able to compile a project to generate the documentation. If it cannot compile it, it will generate the docs up to the point it can compile.

Assuming we have a project that internally uses experimental features, with the current implementation that always passes `&[]` for `experimental` and `no-experimental` that project will never compile. Means the docs will either not be generated or be only partially generated.

The solution is to pass the `(no-)experimental` flags to the `forc doc` as well. Looking at the `forc doc --help` I see it already has the `--no-encoding-v1` flag which makes sense. That should then be removed and replaced with `(no)-experimental`.

Reply:
Hmm, another thought. `forc doc` will still pick the flags if they are declared in `toml`/env. Only the command line ones will not be available. This will very likely be fine for usual case. Perhaps the optimistic look is ok and we don't necessarily extend the CLI or do it in another step.

But removing  the `--no-encoding-v1` still makes sense. And adding flags then perhaps shouldn't be much effort.

Reply:
Done.